Modules/Product/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,1821 @@
+                            }
+                            $Imagedd = new ProductImage();
+                            $Imagedd->product_id = $last_product_id;
+                            $Imagedd->images = $img['src'];
+                            $Imagedd->feature_key = $feature_key;
+                            $Imagedd->save();
+                        }
+                    }
+
+
+                    $variations = $product['variations'];
+                    if (!empty($variations)) {
+                        foreach ($variations as $vars) {
+
+
+                            $variant_key = 'v_' . Str::lower(Str::random(10));
+                            $productvariation = new ProductVariation();
+                            $productvariation->variant_key = $variant_key;
+                            $productvariation->image = $vars['image'][0]['src'];
+                            $productvariation->product_id = $last_product_id;
+                            $productvariation->price = 0;
+                            $productvariation->options1 = $vars['attributes'][0]['name'] ?? '';
+                            $productvariation->options2 = $vars['attributes'][1]['name'] ?? '';
+                            $productvariation->options3 = $vars['attributes'][2]['name'] ?? '';
+                            $productvariation->sku = $vars['sku'];
+                            $productvariation->value1 = $vars['attributes'][0]['option'] ?? '';
+                            $productvariation->value2 = $vars['attributes'][1]['option'] ?? '';
+                            $productvariation->value3 = $vars['attributes'][2]['option'] ?? '';
+                            $productvariation->website_product_id = $product['id'];
+                            $productvariation->website = $user_query_uri;
+                            $productvariation->stock = $vars['stock_quantity'];
+                            $productvariation->variation_id = $vars['id'];
+                            $productvariation->save();
+                        }
+                    }
+                }
+                $Brandstore = new Brandstore();
+                $Brandstore->brand_id = $request->user_id;
+                $Brandstore->website = $user_query_uri;
+                $Brandstore->api_key = $consumer_key;
+                $Brandstore->api_password = $consumer_secret;
+                $Brandstore->types = 'wordpress';
+                $Brandstore->save();
+                $response = ['res' => true, 'msg' => "Successfully Imported", 'data' => ""];
+                Redis::connection();
+                Redis::flushDB();
+                $response = ['res' => true, 'msg' => "Imported Successfully", 'data' => ""];
+            } else {
+                $response = ['res' => false, 'msg' => "Enter valid information", 'data' => ""];
+            }
+        }
+
+        return response()->json($response);
+    }
+
+    public function ImportShopify(Request $request)
```

Comment:
Shopify functionality should be created in its own module to make the functionality modular. This is not the right approach.